sql/parser/eval.go

```diff
@@ -259,6 +263,156 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// maxTime and minTime are not the theoretical max/min for the time.Time
```

Comment:
There are some dragons in the implementation of `time.Time`, but I don't think they break the assertion of my previous comment. All that matters is that after some bounded number of calls to `Add()`, `time.Time` will overflow. I think this is sufficient proof: http://play.golang.org/p/OOVBO7ViWn